SNOW-652526: update release notes for v0.9.0 (#456)
snowflakedb/snowpark-python: CHANGELOG.md

# Release History
## 0.9.0 (2022-08-31)

### New Features:
- Added support for displaying source code as comments in the generated scripts when registering UDFs.

> This feature is turned on by default. To turn if off, pass the new keyword argument `source_code_display` as `False` when calling `register()` or `@udf()`.

Reviewer: if -> it

> - Added support for `column_names` in `df.write.save_as_table()`. You can now specify the names of columns when appending new data to an existing table.

Reviewer 2: This is not accurate. A new param `column_order` is added. When it's "name" and `save_as_table` is in append mode, the dataframe data is appended to an existing table by matching the dataframe's column names and the table's column names. When it's "index", it works as before.

Author: my fault, I read into the analyzer part /facepalm/.

I have updated to `Added keyword argument `column_order` to `df.write.save_as_table()` to specify the matching rules when inserting data into table in append mode.`

how does it look to you now?

- Added function `get_active_session()` in module `snowflake.snowpark.context` to get the current active Snowpark session.
- Added keyword argument `overwrite` to `session.write_pandas()` to allow overwriting contents of a Snowflake table with that of a Pandas DataFrame.

> - Added `FileOperation.put_stream()` to upload local files to a stage via file stream.

Reviewer 3: I would have all of these start with "Added support for" so that the wording is consistent

Author: I feel "added support for" might not be the best for all cases if we're simply adding adding argument/method/functions.

But I got your point on consistency, I have tweaked a bit to make them consistent, how do you like it now?

- Added support for calling table functions from `DataFrame.select()`, `DataFrame.with_column()` and `DataFrame.with_columns` which now take parameters of type `table_function.TableFunctionCall` for columns.

> - Added `TableFunctionCall.alias` to allow aliasing the names of columns that come from the output of table function joins.

Reviewer 4: ```suggestion
- Added `TableFunctionCall.alias()` and `TableFunctionCall.as_()` to allow aliasing the names of columns that come from the output of table function joins.
```

### Bug Fixes:

> - Fixed a bug in which batch insert should not raise error when `statement_params` is not passed to the function.

Reviewer: "raise error" -> "raise an error"

> - Fixed a bug in which column names should be quoted when `create_dataframe` is called with dicts and a given schema.

Reviewer 4: ```suggestion
- Fixed a bug in which column names should be quoted when `session.create_dataframe` is called with dicts and a given schema.
```

> - Fxied a bug in which creation of table shall be skipped if the table already exists and is in append mode.

Reviewer 4: ```suggestion
- Fixed a bug in which creation of table shall be skipped if the table already exists and is in append mode.
```

Reviewer: Also, I suggest changing "shall be skipped" to "is skipped".

Reviewer 2: Is this for `copy_into_table`? Better explicitly mention the API

Author: updated, from the commit: https://github.com/snowflakedb/snowpark-python/commit/1d61cfab560a1381cb49cdad6693f0ea613dc0a5 it's the `df.write.save_as_table` api.

@sfc-gh-sfan , is ther other api impacted?

> - Fxied a bug that packages with underscores cannot be added when registering UDFs.

Reviewer 4: ```suggestion
- Fixed a bug that third-party packages with underscores cannot be added when registering UDFs.
```

Reviewer: Also, change "that" to "where" or "in which".

### Improvements:

> - Improved function `uniform` to infer the types of inputs `max_` and `_min` and cast the limits to `IntegerType` or `FloatType` correspondingly.

Reviewer 4: ```suggestion
- Improved function `functions.uniform()` to infer the types of inputs `max_` and `min_` and cast the limits to `IntegerType` or `FloatType` correspondingly.
```
